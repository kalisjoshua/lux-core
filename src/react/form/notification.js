import React from 'react'; // `React` must be in scope when using JSX

import registry from '../../lib/componentRegistry';

const checkMark =
  `<svg class="icon icon--big" viewBox="0 0 100 100">
    <use xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="#check"></use>
  </svg>`;

const closeMark =
  `<svg class="icon icon--big" viewBox="0 0 100 100">
    <use xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="#times"></use>
  </svg>`;

function NotifyComponent(props) {
  const { message, type } = {
    message: false,
    type: false,
    ...props,
  };

  return (!message || !type)
    ? null
    : (
      <div className="notification">
        <div
          className={`notification__icon notification__icon--${type}`}
          // eslint-disable-next-line react/no-danger
          dangerouslySetInnerHTML={{ __html: checkMark }}
        />

        <div className="notification__text">
          <span>{message}</span>
        </div>

        <div
          className="notification__icon notification__icon--dismiss"
          // eslint-disable-next-line react/no-danger
          dangerouslySetInnerHTML={{ __html: closeMark }}
          data-js-dismiss
        />
      </div>
    );
}
NotifyComponent.propTypes = {
  message: React.PropTypes.node,
  type: React.PropTypes.string,
};

registry('Lux.Form.Notification', NotifyComponent, false);
