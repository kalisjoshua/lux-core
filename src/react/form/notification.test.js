import React from 'react';
import renderer from 'react-test-renderer';

import registry from '../../lib/componentRegistry';

import './notification';

const NotificationComponent = registry('Lux.Form.Notification');

describe('NotificationComponent', function () {
  it('should exist; and should be a function', function () {
    expect(typeof NotificationComponent).toBe('function');
  });
});
