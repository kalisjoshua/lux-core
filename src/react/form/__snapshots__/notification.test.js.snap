// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationComponent should render a "Alert" notification component 1`] = `
<div
  className="notification"
>
  <div
    className="notification__icon notification__icon--alert"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<svg class=\\"icon icon--big\\" viewBox=\\"0 0 100 100\\">
          <use xmlns:xlink=\\"http://www.w3.org/1999/xlink\\" xlink:href=\\"#bell\\"></use>
        </svg>",
      }
    }
  />
  <div
    className="notification__text"
  >
    <span>
      Alert
    </span>
  </div>
  <div
    className="notification__icon notification__icon--dismiss clickable"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<svg class=\\"icon icon--big\\" viewBox=\\"0 0 100 100\\">
          <use xmlns:xlink=\\"http://www.w3.org/1999/xlink\\" xlink:href=\\"#times\\"></use>
        </svg>",
      }
    }
    data-js-notification="dismiss"
    onClick={[Function]}
  />
</div>
`;

exports[`NotificationComponent should render a "Fail" notification component 1`] = `
<div
  className="notification"
>
  <div
    className="notification__icon notification__icon--failure"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<svg class=\\"icon icon--big\\" viewBox=\\"0 0 100 100\\">
          <use xmlns:xlink=\\"http://www.w3.org/1999/xlink\\" xlink:href=\\"#exclamation\\"></use>
        </svg>",
      }
    }
  />
  <div
    className="notification__text"
  >
    <span>
      Fail
    </span>
  </div>
  <div
    className="notification__icon notification__icon--dismiss clickable"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<svg class=\\"icon icon--big\\" viewBox=\\"0 0 100 100\\">
          <use xmlns:xlink=\\"http://www.w3.org/1999/xlink\\" xlink:href=\\"#times\\"></use>
        </svg>",
      }
    }
    data-js-notification="dismiss"
    onClick={[Function]}
  />
</div>
`;

exports[`NotificationComponent should render a "Success" notification component 1`] = `
<div
  className="notification"
>
  <div
    className="notification__icon notification__icon--success"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<svg class=\\"icon icon--big\\" viewBox=\\"0 0 100 100\\">
          <use xmlns:xlink=\\"http://www.w3.org/1999/xlink\\" xlink:href=\\"#check\\"></use>
        </svg>",
      }
    }
  />
  <div
    className="notification__text"
  >
    <span>
      Success
    </span>
  </div>
  <div
    className="notification__icon notification__icon--dismiss clickable"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<svg class=\\"icon icon--big\\" viewBox=\\"0 0 100 100\\">
          <use xmlns:xlink=\\"http://www.w3.org/1999/xlink\\" xlink:href=\\"#times\\"></use>
        </svg>",
      }
    }
    data-js-notification="dismiss"
    onClick={[Function]}
  />
</div>
`;
